python/plugins/processing/ProcessingPlugin.py

```diff
@@ -93,11 +99,11 @@ def initGui(self):
         self.iface.registerMainWindowAction(self.historyAction, 'Ctrl+Alt+H')
         self.menu.addAction(self.historyAction)
 
-        self.resultsAction = QAction(
-            QIcon(os.path.join(cmd_folder, 'images', 'results.svg')),
-            self.tr('&Results Viewer...'), self.iface.mainWindow())
+        self.resultsAction = self.resultsDock.toggleViewAction()
         self.resultsAction.setObjectName('resultsAction')
-        self.resultsAction.triggered.connect(self.openResults)
+        self.resultsAction.setIcon(
```

Comment:
I suggest moving this svg out of the processing folder and across to images/themes, and then using QgsApplication.getThemeIcon() instead (see https://github.com/qgis/QGIS/commit/04eb29336288caf3ba9a858c23969f2f46cec3cb#diff-bca15e5fc8dbdd7d96d60f781fe536c3 for a related commit). This has two advantages:
1. makes the icon reusable outside of processing
2. there's an inbuilt caching logic in getThemeIcon which can speed things up when the same icon is used multiple times.